packages/kbn-config/src/config_service.ts

```diff
@@ -209,10 +209,31 @@ export class ConfigService {
       throw new Error(`No validation schema has been defined for [${namespace}]`);
     }
 
-    const validatedConfig = hasSchema
-      ? await this.atPath<{ enabled?: boolean }>(path).pipe(first()).toPromise()
+    let validatedConfig = hasSchema
+      ? await firstValueFrom(
+          this.getValidatedConfigAtPath$(
+            path,
+            // At this point we don't care about how valid the config is: we just want to read `enabled`
+            { stripUnknownKeys: true }
+          ) as Observable<{ enabled?: boolean }>,
+          { defaultValue: undefined }
+        )
       : undefined;
 
+    // Special use case: when the provided config includes `enabled` and the validated config doesn't,
```

Comment:
This is only at the top level of the object?

Also could we add a test case for this?

Reply:
This is the existing test that reminded me that I would introduce a regression with simply using `stripUnknownKeys: true` above: https://github.com/afharo/kibana/blob/config/strip-unknown-on-serverless/packages/kbn-config/src/config_service.test.ts#L383-L405